src/components/map.jsx

```diff
@@ -23,21 +23,52 @@ let mapProp = {
 // this.ref is referencing the DOM element(used instead of getElement)
 this.mapObj = new google.maps.Map(this.refs.myMap, mapProp);
 
+let submit = document.getElementById('submit');
+let geocoder = new google.maps.Geocoder();
+
+submit.addEventListener('click', ()=>{
+this.geocodeAddress(geocoder, this.mapObj);
+});
+
+
+};
+
+
+geocodeAddress = (geocoder, resultsMap) =>
+{
+let address = document.getElementById('cityInput').value;
+      geocoder.geocode({'address': address}, function(results, status) {
+        if (status === google.maps.GeocoderStatus.OK) {
+          resultsMap.setCenter(results[0].geometry.location);
+          let marker = new google.maps.Marker({
+            map: resultsMap,
+            position: results[0].geometry.location
```

Comment:
What might be a better idea is if we store the location in the reducer, I think there is already an action that is doing this